spec/kit_spec.rb

```diff
@@ -0,0 +1,30 @@
+require_relative '../lib/all'
+
+class HTTPClientDownloadNeededPart < StandardError; end
```

Comment:
This is not necessary; check https://docs.ruby-lang.org/en/3.3/Kernel.html#method-i-catch